trunk/source/PEAtoBoogie/src/de/uni_freiburg/informatik/ultimate/pea2boogie/VerificationResultTransformer.java

```diff
@@ -186,6 +205,17 @@ public IResult convertTraceAbstractionResult(final IResult result) {
 			final String failurePath = formatTimeSequenceMap(delta2var2value);
 			return new ReqCheckRtInconsistentResult<>(element, plugin, failurePath);
 		}
+		// If no InvariantResult is present, fall through to generic FailResult
+		if (spec == Spec.REDUNDANCY && invResult != null) {
+			// Annotation needed for the result to know the respective Check
+			reqCheck.annotate(element);
+			final var invariant = (Expression) invResult.getInvariant();
+			// Only works because spec checks a single requirement
```

Comment:
Perhaps add an assert that it is indeed only one?

Reply:
Done like this:

https://github.com/ultimate-pa/ultimate/blob/8ec82ab4f9eb6cb1f92842c0874efbcbe6c77e4f/trunk/source/PEAtoBoogie/src/de/uni_freiburg/informatik/ultimate/pea2boogie/VerificationResultTransformer.java#L215-L218